pkg/image/registry/imagesignature/strategy.go

```diff
@@ -0,0 +1,74 @@
+package imagesignature
+
+import (
+	"crypto/sha1"
+	"fmt"
+
+	"github.com/golang/glog"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	"github.com/openshift/origin/pkg/image/api/validation"
+)
+
+// strategy implements behavior for ImageStreamTags.
+type strategy struct {
+	runtime.ObjectTyper
+}
+
+var Strategy = &strategy{
+	ObjectTyper: kapi.Scheme,
+}
+
+func (s *strategy) NamespaceScoped() bool {
+	return false
+}
+
+func (s *strategy) PrepareForCreate(obj runtime.Object) {
+	signature := obj.(*imageapi.ImageSignature)
+
+	signature.Conditions = nil
+	signature.ImageIdentity = ""
+	signature.SignedClaims = nil
+	signature.Created = nil
+	signature.IssuedBy = nil
+	signature.IssuedTo = nil
+
+	// generate signature name if it contains just image name
+	imageName, sigName, _ := imageapi.SplitImageSignatureName(signature.Name)
```

Comment:
Do you refer to https://github.com/openshift/origin/pull/9181/files#diff-fba43349a7c241693d9910314a2864abR84 ?